codelists/codeset.py

```diff
@@ -191,3 +195,24 @@ def update(self, updates):
 
         updated_code_to_status = {**self.code_to_status, **code_to_status_updates}
         return type(self)(updated_code_to_status, self.hierarchy)
+
+    def diff(self, other_codeset):
+        """
+        Compare this codeset to another, and return a dict of codes that have been
+        added, changed and removed in this codeset
+        """
+        removed_codes = set(other_codeset.code_to_status) - set(self.code_to_status)
+        all_changes = set(self.code_to_status.items()) - set(
+            other_codeset.code_to_status.items()
+        )
+        new_code_statuses = {
+            (code, status)
+            for code, status in self.code_to_status.items()
+            if other_codeset.code_to_status[code] == "?"
+        }
```

Comment:
If `code` is not in `other_codeset`, we get a `KeyError`.  Could this happen?

At the moment, this method is tested very indirectly, and I'd like to see it thoroughly tested directly.  Not only would this give us more confidence that it's correct, but this would also provide helpful clues about how we expect it to work.